js/assessments/wordComplexityAssessment.js

```diff
@@ -37,14 +42,45 @@ var calculateComplexity = function( wordCount, tooComplexWords, recommendedValue
 		text: i18n.sprintf(
 			i18n.dgettext(
 				"js-text-analysis",
-				// translators: %1$s expands to the percentage of too complex words, %2$d expands to the recommended number of syllables
+				// Translators: %1$s expands to the percentage of too complex words, %2$d expands to the recommended number of syllables
 				// %3$s expands to the recommend maximum
 				"%1$s of the words contain over %2$d syllables, which is more than the recommended maximum of %3$s." ),
 			percentage + "%", recommendedValue, recommendedMaximum + "%" )
 	};
 };
 
 /**
+ * Filters the words that aren't too long.
+ *
+ * @param {Array} words The array with words to filter on complexity.
+ * @returns {Array} The filtered list with complex words.
+ */
+var filterComplexity = function( words ) {
+	return filter( words, function( word ) {
+		return( word.complexity > recommendedValue );
+	} );
+};
+
+/**
+ * Creates markers of words that are complex.
+ *
+ * @param {Paper} paper The Paper object to assess.
+ * @param {Researcher} researcher The Researcher object containing all available researches.
+ * @returns {Array} A list with markers
+ */
+var wordComplexityMarker = function( paper, researcher ) {
+	var wordComplexity = researcher.getResearch( "wordComplexity" );
+	var complexWords = filterComplexity( wordComplexity );
+	return map( complexWords, function( complexWord ) {
+		var marked = marker( complexWord.word );
```

Comment:
This entire line could directly be moved to line 78. Separately defining a variable is unnecessary.